Mathlib/Data/Nat/Digits.lean

```diff
@@ -633,6 +636,22 @@ theorem ofDigits_mod (b k : ℕ) (L : List ℕ) : ofDigits b L % k = ofDigits (b
   ofDigits_modEq b k L
 #align nat.of_digits_mod Nat.ofDigits_mod
 
+theorem ofDigits_mod_eq_head! (b : ℕ) (l : List ℕ) : ofDigits b l % b = l.head! %b := by
+  induction l <;> simp [Nat.ofDigits, Int.ModEq]
```

Comment:
What's the goal after `simp`? Why do you need an extra `rfl`?

Reply:
The goal is 
```
default = 0
```
at that point. Should this be a simp lemma somehow?

Reply:
`Nat.default_eq` or something?

Reply:
I added default_eq to `Mathlib/Init/Data/Nat/Basic.lean`, because I didn't find a better place.
This lead to the removal of `rfl` at other places too.

Reply:
@YaelDillies do you think it makes sense to factor out adding `Nat.default_eq` to a different pull request to make this one easier to review?

Reply:
No, it's fine like that